src/omp/kernels/spectral_processing.f90

```diff
@@ -0,0 +1,99 @@
+module m_omp_spectral
+
+  use m_common, only: dp
+  implicit none
+  
+  contains
+  subroutine process_spectral_div_u(div_u, waves, nx_spec, ny_spec, nz_spec, &
+      y_sp_st, nx, ny, nz, ax, bx, ay, by, az, bz)
+
+    complex(dp), intent(inout), dimension(:, :, :) :: div_u
+    complex(dp), intent(in), dimension(:, :, :) :: waves
+    real(dp), intent(in), dimension(:) :: ax, bx, ay, by, az, bz
+    integer, intent(in) :: nx_spec, ny_spec, nz_spec
+    integer, intent(in) :: y_sp_st
+    integer, intent(in) :: nx, ny, nz
+
+    integer :: i, j, k, ix, iy, iz
+    real(dp) :: tmp_r, tmp_c, div_r, div_c
+
+
+    do k=1, nz_spec
+      do j=1, ny_spec
+        do i=1, nx_spec 
+          ! normalisation
+          div_r = real(div_u(i, j, k), kind=dp)/(nx*ny*nz)
+          div_c = aimag(div_u(i, j, k))/(nx*ny*nz)
```

Comment:
In case the mesh is large, either
```
num / nx / ny / nz
```
or
```
num / (int(nx, int64) * int(ny, int64) * int(nz, int64)
```
to avoid overflows

Reply:
good point, the CUDA implementation might need the same change as I took most of this from there.

Reply:
done for both the cuda and omp backends